src/uct/sm/mm/mm_iface.c

```diff
@@ -227,6 +227,9 @@ static inline ucs_status_t uct_mm_iface_process_recv(uct_mm_iface_t *iface,
         if (status != UCS_OK) {
             /* assign a new receive descriptor to this FIFO element.*/
             uct_mm_assign_desc_to_fifo_elem(iface, elem, 0);
+            if (status != UCS_INPROGRESS) {
```

Comment:
this is not the right fix - in case of error, UCP should release the descriptor and return UCS_OK
AM callback can return either OK or INPROGRESS, not an error.